### ISSUE TITLE
After refactor in version v0.10.0 mime types are often wrong

### ISSUE THREAD

**Issue by spathon** [google-github-actions/upload-cloud-storage#258]
### TL;DR

In version `v0.10.0` all files our files get the mime type `image/svg+xml` in the GCP bucket.

### Expected behavior

Previous version got the correct mime type based on file.

### Observed behavior

All files (.js, .css etc) all get `image/svg+xml` as mime typ in GCP

### Action YAML

```yaml
name: Deploy production

on:
  push:

jobs:
  deploy-production:
    runs-on: ubuntu-latest
    name: Deploy
    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
      - uses: actions/checkout@v2
        with:
          ref: main

      - uses: actions/setup-node@v2
        with:
          node-version: '16.13'

      - name: Install
        run: npm i

      - name: Build
        run: npm run build

      - id: 'auth'
        uses: 'google-github-actions/auth@v0'
        with:
          credentials_json: '${{ secrets.GCP_CREDENTIALS_PRODUCTION }}'

      - id: 'upload-build-folder'
        uses: 'google-github-actions/upload-cloud-storage@v0'
        with:
          path: './build'
          destination: 'path/to/folder'
          parent: false
```

**Comment by karl-run** [comment]
We are also seeing this. :hot_face:

**Comment by cwolf24** [comment]
I have the same for my allure report, where the index.html is now an `application/json` and therefor not viewable in any browser. 😞

**Comment by mvakula** [comment]
Interestingly pointing to specific version like `google-github-actions/upload-cloud-storage@v0.8.0` doesn't fix the issue.

**Comment by JordanDDisch** [comment]
same

**Comment by spathon** [comment]
> Interestingly pointing to specific version like `google-github-actions/upload-cloud-storage@v0.8.0` doesn't fix the issue.

For us `google-github-actions/upload-cloud-storage@v0.9.0` worked

**Comment by mvakula** [comment]
Sorry, for confusion. Locking to specific version like v0.8.0 or v0.9.0 works for us too. There was another unrelated issue.

Btw, looks like locking down CI dependency versions is a good practice 👍

**Comment by karl-run** [comment]
> Btw, looks like locking down CI dependency versions is a good practice +1

Maybe espeically for stuff that is on v0 :weary: Oh well, we learn by breaking production.

**Comment by sileht** [comment]
Same here.

**Comment by sethvargo** [comment]
Hey folks - thanks for bringing this to our attention. I'm looking into this now, but in the meantime, you can pin to `google-github-actions/upload-cloud-storage@0.9.0`.

I will admit that I'm a little confused because I tested that mimeTypes were properly inferred in the refactor for `0.10.0`, and I just pushed up some very specific `Content-Type` tests which are passing.

I'm going to try and reproduce this, but if anyone is able to grab the [step debugging output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging) with `0.10.0` and drop it in a gist/comment, that would be very appreciated!

**Comment by sgd2z** [comment]
Same here everything got deployed as application/node for us and broke web pages.

**Comment by sethvargo** [comment]
Hey folks - we think we identified the root cause and pushed up a fix in #262. Once those tests pass, we'll release v0.10.1.

**Comment by sethvargo** [comment]
Hey folks - I've just released `v0.10.1` (and the tag for `v0` now also points to this new version). Please give it a try and report back. The root cause here is actually a [bug in a downstream client library](https://github.com/googleapis/nodejs-storage/issues/1891) where an object property was being copied onto future uploads.

**Comment by spathon** [comment]
Thanks for the quick fix, it works :)
